internal/util/cache/expirationcache.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2024 Stacklok, Inc.
```

Comment:
I had a look at several but didn't quite find what I was looking for. My initial call was to use [ristretto](https://github.com/dgraph-io/ristretto), but then I figured I could just re-use the implementation we already had, since I really wanted expiration.